enhancements/network/openstack-dual-stack.md

```diff
@@ -0,0 +1,249 @@
+the node addresses will be provided directly by the cloud provider openstack, which can rely on a tag to filter them. This [issue](https://github.com/kubernetes/cloud-provider-openstack/issues/2096) details the idea behind it.
+
+### Implementation Details/Notes/Constraints
+
+The idea behind using bring your own network instead of basic IPI installation is to allow support
+for a wider variety of ways on how to configure the Network and Subnet. As an example, the Ports have
+to be pre-created by the user because when using SLAAC or Stateless the specification of fixed-ips during
+Port creation is not allowed. UPI can also be an option in future releases.
+
+### Risks and Mitigations
+
+We need to validate if the proposal for not setting the node-ip will be a robust enough
+solution that we want to maintain, specially around upgrades. If not robust, we can use the dual-stack support for
+node-ip that is being [developed upstream](https://github.com/kubernetes/kubernetes/pull/116305).
+
+### Drawbacks
+
+NA
+
+## Design Details
+
+### Open Questions
+
+1. Do we want to let the user configure the `MachineConfig` prior installation according
+ to the type of IPv6 mode used or we want to teach the machine-config-operator how to
+ configure those?
+2. How does the routing work when using tenant network and attempt to connect to the API
+ using IPv6 address in the GUA range? until now we tested with Provider network.
+4. Will we restrict the requirement for pre-creation of Port for API and Ingress only to dual-stack?
+5. When to support migration from single stack to dual-stack?
+
+### Test Plan
+
+- Develop CI for basic dual-stack installation with Provider Network.
+
+### Graduation Criteria
+
+The Tech-Preview target is OCP 4.14 and GA would be 4.15.
+
+#### Dev Preview -> Tech Preview
+
+- Succesfully bring up a dual-stack OpenStack cluster without manual intervation
+  of the user with exception of the documented requirements.
+- End user documentation
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- Sufficient time for feedback
+- Migration from single-stack to dual-stack and the other way around
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+
+NA
+
+### Upgrade / Downgrade Strategy
+
+It's not possible to upgrade directly from one version using single-stack to another version with dual-stack.
+The customer would have to first upgrade to a version that has support for dual-stack, but maintaining the single-stack.
+After upgrade the customer could follow [the conversion procedure to dual-stack](https://docs.openshift.com/container-platform/4.12/networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html).
```

Comment:
The upgrade story is probably going to be rewritten (because of https://github.com/openshift/enhancements/pull/1365/files#r1149091298) but just for the sake of being clear - at the moment of writing this comment the enhancement assumes existence of Tags somewhere in the OpenStack infra. Mentioned "conversion procedure to dual-stack" is not covering those, it only covers a scenario where kubelet is ran with `--node-ip`. So if the idea of dropping `--node-ip` stands, the linked doc is not relevant for this platform